controller.h

```diff
@@ -508,7 +508,11 @@ inline void from_json(const nlohmann::json &j, unique_ptr<Controller> & ptrContr
 
         // Extract canvases
         for (const auto &canvasJson : j.at("canvases"))
-            ptrController->AddCanvas(canvasJson.get<unique_ptr<ICanvas>>());
+        {
+            unique_ptr<ICanvas> ptrCanvas;
+            from_json(canvasJson, ptrCanvas);
+            ptrController->AddCanvas(std::move(ptrCanvas));
+        }
```

Comment:
Is there a reason you prefer the more verbose way of doing this, and maybe even more so, specifically in this place?

Reply:
This one too...

```suggestion
            ptrController->AddCanvas(canvasJson.get<unique_ptr<ICanvas>>());
```